### PR TITLE
feat: promql create and skeleton

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

PromQL crate and skeleton.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

Related to #596

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
👍

File: docs/rfc/20221207_promql.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+* Author: Dennis Zhuang
+* Updated: 2022/12/13
+
+## Motivation
+The [PromQL](https://prometheus.io/docs/prometheus/latest/querying/basics/) supported by Prometheus is used widely in cloud-native observability.  We want to enable it natively in the GreptimeDB.
+The GreptimeDB already supports [remote read/write protocol for Prometheus](https://docs.greptime.com/user-guide/supported-protocols/prometheus). But it's not good enough for performance and DevOps. We want to implement PromQL in pure Rust, and cooperate reasonably with our query engine and table engine. We want to push computation down  into storage, reducing data transfer and providing the best performance. And also keep compatible with  Prometheus.
+## Design(draft)
```
Comment (fengjiachun) [inline docs/rfc/20221207_promql.md:7]: Add a new line

**Review by fengjiachun — APPROVED** [review]

**Review by yuanbohan — APPROVED** [review]
LGTM

File: docs/rfc/20221207_promql.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+TODO
+
+### Test
+Prometheus provides a compatible test suite for other implementations. https://github.com/promlabs/promql-compliance-tester
+We can use it to test our implementation.
+
+## Milestones
+
+The first milestone may be in 2023, January. Make the PromQL run and let the compatible test cases pass over 60%.
```
Comment (waynexia) [inline docs/rfc/20221207_promql.md:66]: This doesn't look like a part of RFC... Better to track milestone in corresponding issue

Comment (killme2008) [reply docs/rfc/20221207_promql.md]: Already deleted.

**Pull request merged**